CMakeLists.txt

```diff
@@ -587,6 +588,111 @@ if (BUILD_TESTING)
+        target_include_directories(fuzztest PUBLIC tests)
+        target_link_libraries(fuzztest PUBLIC ${PROJECT_NAME})
+
+        # Set default values for fuzzing if not defined
+        if(NOT DEFINED LIBFUZZER_LIB)
+            set(LIBFUZZER_LIB "/usr/local/libfuzzer/lib/libFuzzer.a")
+        endif()
+        
+        if(NOT DEFINED FUZZ_TIMEOUT_SEC)
+            set(FUZZ_TIMEOUT_SEC 120)
+        endif()
+
+        if(NOT DEFINED CORPUS_UPLOAD_LOC)
+            set(CORPUS_UPLOAD_LOC "none")
+        endif()
+
+        if(NOT DEFINED ARTIFACT_UPLOAD_LOC)
+            set(ARTIFACT_UPLOAD_LOC "none")
+        endif()
+
+        if(NOT DEFINED FUZZ_TESTS)
+            set(FUZZ_TESTS "${TESTS}")
+        endif()
+
+        # Build LD_PRELOAD shared libraries
+        include_directories(
+            ${CMAKE_CURRENT_SOURCE_DIR}
+            ${CMAKE_CURRENT_SOURCE_DIR}/api
+        )
+        set(LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/fuzz/LD_PRELOAD)
+        file(GLOB LIBRARY_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/tests/fuzz/LD_PRELOAD/*.c")
+        foreach(SRC ${LIBRARY_SRCS})
+            get_filename_component(LIB_NAME ${SRC} NAME_WE)
+            add_library(${LIB_NAME} SHARED ${SRC})
+
+            # Set the output directory and remove the default "lib" prefix
+            set_target_properties(${LIB_NAME} PROPERTIES
+                PREFIX ""
+                LIBRARY_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_DIRECTORY}
+            )
+            target_compile_options(${LIB_NAME} PRIVATE
+                -Wno-unreachable-code -O0 -fPIC -Wno-deprecated-declarations -fvisibility=default
+            )
+        endforeach()
+
+        set(CMAKE_C_COMPILER clang)
+
+        foreach(src ${FUZZ_TEST_SRCS})
+            get_filename_component(TEST_NAME ${src} NAME_WE)
+
+            add_executable(${TEST_NAME} ${src})
+            target_include_directories(${TEST_NAME} PRIVATE ./)
+            
+            target_compile_options(${TEST_NAME} PRIVATE 
+                -g -O0
+                -fsanitize-coverage=trace-pc-guard -fsanitize=leak 
+                -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-deprecated -Wno-unused-result
+            )
+            target_link_libraries(${TEST_NAME} PRIVATE 
+                fuzztest
+                ${LIBFUZZER_LIB}  # Manually link old libFuzzer.a
+                -lstdc++
+                -fsanitize-coverage=trace-pc-guard -fsanitize=leak
+            )
+
+            # Set the output directory for the fuzzing binaries
+            set(FUZZ_BIN_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tests/fuzz")
+            set_target_properties(${TEST_NAME} PROPERTIES
+                RUNTIME_OUTPUT_DIRECTORY ${FUZZ_BIN_DIR}
+            )
+
+            list(APPEND EXECUTABLE_TARGETS ${TEST_NAME})
+        endforeach()
+
+        foreach(TEST_NAME ${EXECUTABLE_TARGETS})
+            add_custom_command(
+                OUTPUT ${TEST_NAME}_result
```

Comment:
preference: Maybe name this ${TEST_NAME}_script? In my mental model of make, that seems a bit more straightforward?
```suggestion
        foreach(TEST_NAME ${FUZZ_TEST_EXECUTABLES})
            add_custom_command(
                OUTPUT ${TEST_NAME}_script
```